File: processor/src/test/resources/approvals/beanvalidation/IdsProductIdValueExtractor.java
package test;

import jakarta.validation.valueextraction.ExtractedValue;
import jakarta.validation.valueextraction.ValueExtractor;
import scenarios.java.Ids;

import javax.annotation.processing.Generated;

@Generated("com.qualityminds.lazyval.processor.internal.codegen.beanvalidation.BeanValidationGenerator")
public class IdsProductIdValueExtractor implements ValueExtractor<Ids. @ExtractedValue(type = String.class) ProductId> {
  @Override
  public void extractValues(Ids.ProductId originalValue, ValueExtractor.ValueReceiver receiver) {
    if (originalValue == null) {
      receiver.value(null, null);
      return;
    }
    receiver.value(null, originalValue.value());
  }
}

File: ksp/src/test/resources/approvals/beanvalidation/IdsProductIdValueExtractor.kt
package test

import jakarta.`annotation`.Generated
import jakarta.validation.valueextraction.ValueExtractor
import scenarios.kotlin.Ids

@Generated("com.qualityminds.lazyval.ksp.internal.codegen.beanvalidation.BeanValidationGenerator")
public class IdsProductIdValueExtractor : IdsProductIdValueExtractorBase() {
  override fun extractValues(originalValue: Ids.ProductId?, `receiver`: ValueExtractor.ValueReceiver) {
    if (originalValue == null) {
      receiver.value(null, null)
      return
    }
    receiver.value(null, originalValue.value)
  }
}
